### ISSUE TITLE
object-shorthand fixer outputs invalid syntax if a computed property name is parenthesized

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7744]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
  object-shorthand: error
parserOptions:
  ecmaVersion: 2015
```

**What did you do? Please include the actual source code causing the issue.**

```js
({ [(foo)]: function() {} });
```

**What did you expect to happen?**

I expected ESLint to report an error and fix the code to:

```js
({ [(foo)]() {} });
```

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reported an error and fixed the code to:

```js
({ [(foo)() {} })
```

**Comment by not-an-aardvark** [comment]
Oh also, it looks like there's more fix logic that hasn't accounted for async, so the following will cause syntax errors:

```js
/* eslint object-shorthand: ["error", never] */

({ async foo() {} })

// incorrectly gets fixed to:

({ async foo: function() {} })
```

(The fixes in https://github.com/eslint/eslint/pull/7575 only handled the cases where longform syntax was converted to shorthand syntax.)
